### ISSUE TITLE
Potentially bad logic in `TheoryArithPrivate::selectSimplex` when `useFC()` is false

### ISSUE THREAD

**Issue by andrewvaughanj** [cvc5/cvc5#7037]
**Describe the bug**

I was trying to work out the meaning behind:

- https://github.com/cvc5/cvc5/blob/42b8e0a4fc5b23dfce79cbadc08a8aa8542997e0/src/theory/arith/theory_arith_private.cpp#L2982-L2983

with respect to the difference in behaviour in `cvc5` if you do `--use-approx` without GLPK, and then I came across what looks like a bug in `TheoryArithPrivate::selectSimplex`.

For `pass1`, we have:

- https://github.com/cvc5/cvc5/blob/42b8e0a4fc5b23dfce79cbadc08a8aa8542997e0/src/theory/arith/theory_arith_private.cpp#L2876-L2881

Which results in: `useFC` means we get `d_fcSimplex`, `useSOI` means we get `d_soiSimplex`, otherwise we get `d_dualSimplex`.

So far, so good ...

However, for "not `pass 1`", we have:

- https://github.com/cvc5/cvc5/blob/42b8e0a4fc5b23dfce79cbadc08a8aa8542997e0/src/theory/arith/theory_arith_private.cpp#L2888-L2893

Which results in: `useFC` means we get `d_fcSimplex`, `useSOI` means we get `d_soiSimplex`, otherwise we **also** get `d_soiSimplex` (so the same as `useSOI`).

So there's either a bug here and the "otherwise" case should give `d_dualSimplex` **or** the check against `useSOI` should be removed and then the function should return `d_soiSimplex` as long as we don't have `useFC`.

It is worth pointing out that both `useFC` and `useSOI` default to _false_, so this means that the buggy logic is always exercised when selecting a simplex solver.

This code was written by @timothy-king in 9ccdea06edbc72e3ecd282e9e015f6fc4b2e7173 (from 2014!) and hasn't been touched since.

Does anyone know the intended behaviour of this code? If we know, I'll happily fix it.

**Comment by nafur** [comment]
I've found this thing as well a few months ago. I discussed this with @ajreynol at the time, and we decided to not touch this code unless we make a serious effort at refactoring/modernizing the linear solver.

**Comment by andrewvaughanj** [comment]
Thanks, @nafur! Would it be worth (at all) making that change and then seeing if it affects performance at all?

**Comment by ajreynol** [comment]
I agree with @nafur, my preference would be to not change this unless we plan a major refactor.

**Comment by ajreynol** [comment]
I'm going to move this to cvc5-projects for internal tracking.
